cmd/hook.go

```diff
@@ -376,7 +376,12 @@ Gitea or set your environment appropriately.`, "")
 		oldCommitIDs[count] = string(fields[0])
 		newCommitIDs[count] = string(fields[1])
 		refFullNames[count] = git.RefName(fields[2])
-		if refFullNames[count] == git.BranchPrefix+"master" && newCommitIDs[count] != git.EmptySHA && count == total {
+
+		hash, err := git.HashTypeInterfaceFromHashString(newCommitIDs[count])
+		if err != nil {
+			hash = &git.Sha1Hash{}
+		}
+		if refFullNames[count] == git.BranchPrefix+"master" && newCommitIDs[count] != hash.Empty() && count == total {
```

Comment:
hmm this is outside of this pull scobe ... but we still have this magic value :/

we should try to determine what the default branch of a repo or at least of the system is

Reply:
->  #27497